sdks/python/apache_beam/utils/subprocess_server.py

```diff
@@ -114,17 +114,27 @@ def start_process(self):
         cmd = [arg.replace('{{PORT}}', str(port)) for arg in self._cmd]
       endpoint = 'localhost:%s' % port
       _LOGGER.info("Starting service with %s", str(cmd).replace("',", "'"))
+
+      with tempfile.NamedTemporaryFile(delete=False) as stdout_file:
+        self._stdout_file_name = stdout_file.name
       self._process = subprocess.Popen(
-          cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+          cmd,
+          stdout=open(self._stdout_file_name, 'wb'),
```

Comment:
the object returned by NamedTemporaryFile is already an opened file object with mode `w+b`, if you apply the above suggestion, this could be:
```suggestion
          stdout=self._stdout_file,
```